api/api/serializers.py

```diff
@@ -113,13 +113,28 @@ def create(self, validated_data):
         return spot
 
     def update(self, instance, validated_data):
+        if 'rapport_document' in validated_data and validated_data['rapport_document'] is None:
+            # delete rapport_document
+            self.delete_document(spot=instance, document_type=Document.DocumentType.Rapportage)
+
+        if 'design_document' in validated_data and validated_data['design_document'] is None:
+            # delete design document
+            self.delete_document(spot=instance, document_type=Document.DocumentType.Ontwerp)
+
         rapport_file = validated_data.pop('rapport_document', None)
         design_file = validated_data.pop('design_document', None)
 
         spot = super().update(instance, validated_data)
         self.handle_documents(spot, rapport_file, design_file)
         return spot
 
+    def delete_document(self, spot, document_type):
+        document = Document.objects.filter(spot=spot, type=document_type).first()
```

Comment:
What is the scenario that you will delete a document that doesn't exist in the db?

Reply:
Good point, I added an error log when this occurs.